### ISSUE TITLE
strange <80><fc>^B characters after `csw"` surrounding

### ISSUE THREAD

**Issue by mkreim** [tpope/vim-surround#313]
Hi,

I have a problem with the surround plugin and I have no idea what is causing it or how to debug into it to find the source.

So what happens is:
I have a text like:
```
server
```
I place the cursor somewhere in the word and type `csw"`
and I end with (for a strange reason I cannot copy it therefore the screenshot)
![image](https://user-images.githubusercontent.com/25951670/95970055-ca85d980-0e0f-11eb-8b16-49e3c8fe0ce5.png)

- The problem occurs only in gvim and not in the terminal. 
- I am running ubuntu 20.04 LTS
- and using a self compiled version of vim:
```$ gvim --version
VIM - Vi IMproved 8.2 (2019 Dec 12 kompiliert am Sep 16 2020 10:27:56)
Inklusive der Patches: 1-1690
Übersetzt von Michael
Riesige Version mit GTK3 GUI.
 Ein- (+) oder ausschließlich (-) der Eigenschaften:
+acl               -farsi             +mouse_sgr         +tag_binary
+arabic            +file_in_path      -mouse_sysmouse    -tag_old_static
+autocmd           +find_in_path      +mouse_urxvt       -tag_any_white
+autochdir         +float             +mouse_xterm       -tcl
-autoservername    +folding           +multi_byte        +termguicolors
+balloon_eval      -footer            +multi_lang        +terminal
+balloon_eval_term +fork()            -mzscheme          +terminfo
+browse            +gettext           +netbeans_intg     +termresponse
++builtin_terms    -hangul_input      +num64             +textobjects
+byte_offset       +iconv             +packages          +textprop
+channel           +insert_expand     +path_extra        +timers
+cindent           +ipv6              -perl              +title
+clientserver      +job               +persistent_undo   +toolbar
+clipboard         +jumplist          +popupwin          +user_commands
+cmdline_compl     +keymap            +postscript        +vartabs
+cmdline_hist      +lambda            +printer           +vertsplit
+cmdline_info      +langmap           +profile           +virtualedit
+comments          +libcall           -python            +visual
+conceal           +linebreak         +python3           +visualextra
+cryptv            +lispindent        +quickfix          +viminfo
+cscope            +listcmds          +reltime           +vreplace
+cursorbind        +localmap          +rightleft         +wildignore
+cursorshape       -lua               -ruby              +wildmenu
+dialog_con_gui    +menu              +scrollbind        +windows
+diff              +mksession         +signs             +writebackup
+digraphs          +modify_fname      +smartindent       +X11
+dnd               +mouse             +sound             -xfontset
-ebcdic            +mouseshape        +spell             -xim
+emacs_tags        +mouse_dec         +startuptime       +xpm
+eval              +mouse_gpm         +statusline        +xsmp_interact
+ex_extra          -mouse_jsbterm     -sun_workshop      +xterm_clipboard
+extra_search      +mouse_netterm     +syntax            -xterm_save
          System-vimrc-Datei: "$VIM/vimrc"
        Benutzer-vimrc-Datei: "$HOME/.vimrc"
 zweite Benutzer-vimrc-Datei: "~/.vim/vimrc"
         Benutzer-exrc-Datei: "$HOME/.exrc"
         System-gvimrc-Datei: "$VIM/gvimrc"
       Benutzer-gvimrc-Datei: "$HOME/.gvimrc"
zweite Benutzer-gvimrc-Datei: "~/.vim/gvimrc"
      defaults Datei: "$VIMRUNTIME/defaults.vim"
           System-Menü-Datei: "$VIMRUNTIME/menu.vim"
     Voreinstellung für $VIM: "/home/mkreim/bin/vim8/share/vim"
Übersetzt: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include   -g -O2 -D_REENTRANT -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1       
Linken: gcc   -L/usr/local/lib -Wl,--as-needed -o vim   -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lSM -lICE -lXpm -lXt -lX11 -lXdmcp -lSM -lICE  -lm -ltinfo  -lselinux  -lcanberra -lacl -lattr -lgpm -ldl     -L/usr/lib/python3.8/config-3.8-x86_64-linux-gnu -lpython3.8 -lcrypt -lpthread -ldl -lutil -lm -lm 
```

Do you have any ideas what this is causing it? Is this a strange utf-8 problem with the gui vim version? 

Thanks a lot

**Comment by MyBeansAreBaked** [comment]
I also experience this issue using neovim in the foot terminal on archlinux. Strangely, it isn't a problem in alacritty.

**Comment by MyBeansAreBaked** [comment]
@dnkl hanks for the info. Downgrading to neovim v0.6.1 does fix this issue in the stable foot release. Upgrading to the git version of foot (`1.11.0-164-gdd03e10c`) also fixes this issue, on neovim v0.7.0.

I suppose that this original ticket must have been to do with an issue in gvim.

**Comment by dnkl** [comment]
Fixed in [foot-1.12.0](https://codeberg.org/dnkl/foot/releases/tag/1.12.0)

**Comment by tpope** [comment]
The original issue was about GVim on Ubuntu. I'm not seeing it on the new 22.04 LTS, so I'm going to chalk this up as another case of a fixed bug on other software, until I hear otherwise.
